### PR TITLE
Add benchmark regression runner

### PR DESCRIPTION
**Pull request by bf4**
Per https://github.com/rails-api/active_model_serializers/issues/1575

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
Do you plan on adding docs and examples?

**Comment by NullVoxPopuli** [comment]
I mean, in the docs/ folder. I see you have in-line docs here. :-)

File: bin/bench_regression [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+    _    = args.shift # remove 'current' from args
+    cmd  = args
+    benchmarking.run_benchmark(cmd)
+  when 'revisions'
+    # Runs on every revision
+
+    # super simple command line parsing
+    args = ARGV.dup
+    _         = args.shift
+    ref1      = args.shift # remove 'revisions' from args
```
Comment (groyoh) [inline bin/bench_regression]: Shouldn't the comment be on the line above?

File: bin/bench_regression [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+      `git log --oneline -1 #{rev}`.chomp
+    end
+
+    def revisions(start_ref, end_ref)
+      cmd = "git rev-list --reverse #{start_ref}..#{end_ref}"
+      `#{cmd}`.chomp.split("\n")
+    end
+
+    def checkout_ref(ref)
+      `git checkout #{ref}`.chomp
```
Comment (groyoh) [inline bin/bench_regression]: This should probably be muted. Here is what I get as output:

```sh
yor@localhost ~/R/g/active_model_serializers> bundle exec ./bin/bench_regression  revisions 146968d6586f44810be1898f4962a666880558de master
Checking out: 1d84892 Never mutate controller options
Note: checking out '1d848922768e163e57e584a9c524c8d470280aba'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 1d84892... Never mutate controller options
```

**Comment by groyoh** [comment]
This looks good to me. We can create a follow up PRs afterward and adds docs there. Otherwise this PRwill stay open for a bit.

**Pull request merged**